testsuite/standalone/src/test/java/org/wildfly/extras/creaper/core/online/SslOnlineManagementClientTest.java

```diff
@@ -0,0 +1,157 @@
+
+        prepareKeyAndTrustStoreFiles();
+        prepareSslOptions();
+        configureServerManagementInterfaces();
+    }
+
+    @AfterClass
+    public static void restoreConfiguration() throws CommandFailedException {
+        offlineClient.apply(CONFIGURATION_BACKUP.restore());
+    }
+
+    @AfterClass
+    public static void deleteKeyAndTrustStoreFiles() {
+        serverKeyStoreFile.delete();
+        serverTrustStoreFile.delete();
+        clientKeyStoreFile.delete();
+        clientTrustStoreFile.delete();
+    }
+
+    @Before
+    @Override
+    public void connect() throws IOException {
+        if (this.controller.isStarted(ManualTests.ARQUILLIAN_CONTAINER_MGMT_PROTOCOL_REMOTE)) {
+            client = ManagementClient.online(OnlineOptions.standalone().localDefault().ssl(sslOptions).build());
+        }
+    }
+
+    @Test
+    @InSequence(Integer.MIN_VALUE)
+    public void startServer() throws IOException, CommandFailedException, GeneralSecurityException {
+        this.controller.start(ManualTests.ARQUILLIAN_CONTAINER_MGMT_PROTOCOL_REMOTE);
+    }
+
+    @Test
+    @InSequence(Integer.MAX_VALUE)
+    public void stopServer() throws IOException, CommandFailedException {
+        this.controller.stop(ManualTests.ARQUILLIAN_CONTAINER_MGMT_PROTOCOL_REMOTE);
+    }
+
+    @Test(expected = IllegalStateException.class)
+    @Override
+    public void clientConfiguredForDomain() throws IOException {
+        ManagementClient.online(OnlineOptions.domain().build().localDefault().ssl(sslOptions).build());
+    }
+
+
+    private static void prepareKeyAndTrustStoreFiles() throws GeneralSecurityException, IOException {
+        final KeyPairAndCertificate serverKeyPair = KeyPairAndCertificate.generateSelfSigned("test-creaper-server");
+        serverKeyStoreFile = serverKeyPair.toTmpKeyStoreFile(SERVER_KEY_ALIAS, SERVER_KEY_PASSWORD);
+        clientTrustStoreFile = serverKeyPair.toTmpTrustStoreFile(SERVER_CERTIFICATE_ALIAS, SERVER_CERTIFICATE_PASSWORD);
+
+        final KeyPairAndCertificate clientKeyPair = KeyPairAndCertificate.generateSelfSigned("test-creaper-client");
+        clientKeyStoreFile = clientKeyPair.toTmpKeyStoreFile(CLIENT_KEY_ALIAS, CLIENT_KEY_PASSWORD);
+        serverTrustStoreFile = clientKeyPair.toTmpTrustStoreFile(CLIENT_CERTIFICATE_ALIAS, CLIENT_CERTIFICATE_PASSWORD);
+    }
+
+    private static void prepareSslOptions() {
+        sslOptions = new SslOptions.Builder()
+                .keyStore(clientKeyStoreFile)
+                .keyStorePassword(CLIENT_KEY_PASSWORD)
+                .key(CLIENT_KEY_ALIAS, CLIENT_KEY_PASSWORD)
+                .trustStore(clientTrustStoreFile)
+                .trustStorePassword(SERVER_CERTIFICATE_PASSWORD)
+                .build();
+    }
+
+    private static void configureServerManagementInterfaces() throws CommandFailedException {
+        AddHttpsManagementSecurityRealm addHttpsSecurityRealm = new AddHttpsManagementSecurityRealm
+                .Builder(SECURITY_REALM)
```

Comment:
`.Builder(SECURITY_REALM)` should be at the end of previous line. I know this looks slightly better in the chain, but having the constructor invocation expression split into two lines is just weird.